rclpy/rclpy/__init__.py

```diff
@@ -14,25 +14,29 @@
 
 import sys
 
-from rclpy.executors import SingleThreadedExecutor as _SingleThreadedExecutor
-from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
-from rclpy.node import Node
 from rclpy.utilities import get_rmw_implementation_identifier  # noqa
 from rclpy.utilities import ok
 from rclpy.utilities import shutdown  # noqa
 from rclpy.utilities import try_shutdown  # noqa
 
 
 def init(*, args=None):
-    return _rclpy.rclpy_init(args if args is not None else sys.argv)
+    # imported locally to avoid loading extensions on module import
+    from rclpy.impl.implementation_singleton import rclpy_implementation
+    return rclpy_implementation.rclpy_init(
+        args if args is not None else sys.argv)
 
 
 def create_node(node_name, *, namespace=None):
+    # imported locally to avoid loading extensions on module import
+    from rclpy.node import Node
```

Comment:
Is delaying the import a necessary change with the new approach, or an additional feature?

Reply:
It is necessary since the `node` module already imports the singleton module. But we need to delay the instantiation of the singletons until we had a chance to override the `_import` functions.